alts/src/main/java/io/grpc/alts/InternalCheckGcpEnvironment.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.alts;
+
+import io.grpc.Internal;
+
+/**
+ * Internal accessor for {@link CheckGcpEnvironment}. This is intended for usage internal to
+ * the gRPC team. If you *really* think you need to use this, contact the gRPC team first.
+ */
+@Internal
+public final class InternalCheckGcpEnvironment {
```

Comment:
The pattern that was used elsewhere was _necessary_ because those APIs are in `internal` package, and need an `@Internal` public accessor in the main package. This one is already in the main package, it can just be renamed to an `@Internal` public class without an accessor.

Reply:
> The pattern that was used elsewhere was necessary because those APIs are in internal package, and need an @Internal public accessor in the main package. 

That's not a solid reason.

Some examples are:
- [`InternalCensusstatsAccessor`](https://github.com/grpc/grpc-java/blob/master/census/src/main/java/io/grpc/census/InternalCensusStatsAccessor.java) for accessing `CensusStatsModule` (same for tracing).
- [`InternalCallOptions`](https://github.com/grpc/grpc-java/blob/master/api/src/main/java/io/grpc/InternalCallOptions.java) for accessing the `getWaitForReady()` API.
- [`InternalClientCalls`](https://github.com/grpc/grpc-java/blob/master/stub/src/main/java/io/grpc/stub/InternalClientCalls.java) for accessing stubtype APIs.

The real reason should be, does it make sense to make `isOnGcp()` a public API (included in Javadoc) for gRPC users? Marking it `@Internal` is far from enough. Actually, only annotating with `@Internal` while exposing it to users in documentation does nothing.

Reply:
Renaming the current `CheckGcpEnvironment` (only one API there) directly to `@Internal public class InternalCheckGcpEnvironment` is all that we need. No need to keep a package private copy and a public accessor.

Reply:
Oh yeah, that's a fair point. No need the extra wrapping. Fixed.